### PR TITLE
RBAC Roles

### PR DESCRIPTION
**Pull request by newswangerd**
This is a mega PR to merge all of the work that has been done in the `feature/rbac-roles` branch.

### CODE REVIEW

File: galaxy_ng/app/tasks/registry_sync.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ def launch_container_remote_sync(remote, registry, repository):
             "remote_pk": str(remote.pk),
             "repository_pk": str(repository.pk),
             "mirror": True,
+            "signed_only": False,
```
Comment (alikins) [inline galaxy_ng/app/tasks/registry_sync.py:21]: Not sure I understand how this is related to RBAC?

File: galaxy_ng/app/migrations/0028_move_perms_to_roles.py [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+def does_table_exist(table_name):
+    return table_name in connection.introspection.table_names()
+
+
+def migrate_group_permissions_to_roles(apps, schema_editor):
+    """
+    Migration strategy:
+    - Apply locked roles to the group that match the group's permission set.
+    - If any permissions are left over after a role is applied add a role with a single
+      permission to it to make up for each missing permission.
```
Comment (alikins) [inline galaxy_ng/app/migrations/0028_move_perms_to_roles.py:307]: This file needs some whitespace cleanup.

File: galaxy_ng/tests/unit/migrations/test_0028_move_perms_to_roles.py [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+    _assign_perm = None
+
+    def _get_permission(self, permission_name):
+        app_label = permission_name.split('.')[0]
+        codename = permission_name.split('.')[1]
+        return Permission.objects.get(
+            content_type__app_label=app_label,
+            codename=codename
+        )
+
```
Comment (alikins) [inline galaxy_ng/tests/unit/migrations/test_0028_move_perms_to_roles.py:186]: Some whitespace cleanup needed here.

File: requirements/requirements.common.txt [hunk 4]
```diff
@@ -1,150 +1,83 @@
...
-    # via pulpcore
-django-filter==21.1
-    # via pulpcore
-django-guardian==2.4.0
-    # via pulpcore
-django-guid==3.2.2
-    # via pulpcore
-django-import-export==2.7.1
-    # via pulpcore
 django-ipware==3.0.7
```
Comment (alikins) [inline requirements/requirements.common.txt:63]: (compliment) A++ :100: Nice to see some deps go away. Nice Work.

File: setup.py [hunk 5]
```diff
@@ -86,13 +86,35 @@ def run(self):
...
+    """
+    if specifier:
+        return plugin + specifier
+    else:
+        repo = plugin.replace("-", "_")
+        return (
+            f"{plugin}@git+https://git@github.com/"
+            f"{gh_namespace}/{repo}.git@{ref}#egg={plugin}"
+        )
+
```
Comment (alikins) [inline setup.py:109]: Looks very useful. But it could probably be split into it's own PR.

File: template_config.yml [hunk 6]
```diff
@@ -103,3 +107,4 @@ travis_notifications: null
 update_github: false
 upgrade_range: []
 use_issue_template: false
+
```
Comment (alikins) [inline template_config.yml:110]: Trailing newline needs cleanup

**Review by alikins — CHANGES_REQUESTED** [review]
Mostly small stuff, but a few things that I'd like to see. The stuff that can be split into a smaller PR for example.

File: .github/stale.yml [hunk 7]
```diff
@@ -0,0 +1,53 @@
+# Configuration for probot-stale - https://github.com/probot/stale
```
[the remark this answers is not in the document]
Comment (newswangerd) [reply .github/stale.yml:1]: this came with the plugin template

File: .github/workflows/scripts/before_install.sh [hunk 8]
```diff
@@ -111,30 +111,36 @@ fi
...
   git checkout $PULP_ANSIBLE_PR_NUMBER
 fi
 
 cd ..
 
-git clone --depth=1 https://github.com/pulp/pulp_container.git --branch 2.10.2
+git clone --depth=1 https://github.com/pulp/pulp_container.git
 cd pulp_container
 
+git fetch --depth=1 origin 20bec438da54ddd6da4102f046f12c9f796a2ce9
```
[the remark this answers is not in the document]
Comment (newswangerd) [reply .github/workflows/scripts/before_install.sh]: plugin template updates

File: .github/stale.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (newswangerd) [reply .github/stale.yml:1]: plugin template updates

[the remark this answers is not in the document]
Comment (newswangerd) [reply .github/stale.yml:1]: plugin template updates

File: galaxy_ng/app/api/urls.py [hunk 9]
```diff
@@ -41,14 +41,14 @@
...
-         name="root"),
+         name="content-root"),
 
     # This can be removed when ansible-galaxy stops appending '/api' to the
     # urls.
     path("content/<str:path>/api/",
          views.ApiRedirectView.as_view(),
          name="api-redirect",
-         kwargs={"reverse_url_name": "galaxy:api:content:root"}),
+         kwargs={"reverse_url_name": "galaxy:api:v3:content-root"}),
```
[the remark this answers is not in the document]
Comment (newswangerd) [reply galaxy_ng/app/api/urls.py:51]: i think this has to do with upgrading pulp

**Review by awcrosby — APPROVED** [review]

**Comment by newswangerd** [comment]
Moved to #1413 since the PR check seems to be permanently broken here.

**Pull request closed without merging**